### PR TITLE
Added ERC1155, renamed claim to withdraw, show none on address page, ...

### LINKED ISSUE #183: Add ERC1155 NFT token support
# Description

Add ERC1155 to Teloscan, like ERC721 & ERC20 implementations

# Tasks

- [ ] Show transfers on transaction page's general tab
- [ ] Show transfers on transaction page's logs page
- [ ] Include the ERC1155 transfers to the NFTs
- [ ] Rename the tab to include ERC1155
- [ ] Add a badge to indicate if the transaction is ERC1155 or not
- [ ] Show a thumbnail image next to the transaction

### PR DESCRIPTION
**Pull request by tomtomcrypto**
# Fixes #183 #233 #232 #238 #246 

## Description

This adds ERC115 support to teloscan as well as various UI fixes & additions

## Tasks

- [x] #183
- [x] #233
- [x] #232
- [x] #238
- [x] #246

## Test scenarios

- [x] User should be able to see an ERC1155 transfer on the transaction page and in its log
- [x] User should see an ERC1155 tab when navigating to the address page
- [x] User should be able to navigate to the ERC1155 tab of the address page and see a list of all their ERC1155 transfers
- [x] User should see a "Withdraw" tab instead of a "Claim" tab on the staking page
- [x] All mention of claim should be replaced by withdraw in the "Claim" tab
- [x] Claim component should be renamed to Withdraw
- [x] User should be able to see the current nonce of an address on the address page
- [x] Show "Contract deployment" instead of "TLOS Transfer" for contract creation internal trx
- [x] Show an error message on staking page if user is logged in with a native wallet (no signers)

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

File: src/components/TransferTable.vue [hunk 1]
```diff
@@ -149,7 +157,11 @@ export default {
                                 valueDisplay = 'Unknown precision';
                             }
                         } else {
-                            valueDisplay = `Id #${parseInt(log.topics[3], 16)}`;
+                            let tokenId = (this.tokenType === 'erc1155') ? BigNumber.from(log.data.substr(0, 66)).toString() : BigNumber.from(log.topics[3]).toString();
```
Comment (donnyquixotic) [inline src/components/TransferTable.vue:160]: 'magic numbers' that might benefit from being a descriptive const variable for readability

**Review by donnyquixotic — CHANGES_REQUESTED** [review]

File: src/components/TransferTable.vue [hunk 2]
```diff
@@ -125,13 +128,18 @@ export default {
...
                         const address = `0x${log.address.substring(log.address.length - 40)}`;
-                        const from = `0x${log.topics[1].substring(log.topics[1].length - 40)}`;
-                        const to = `0x${log.topics[2].substring(log.topics[2].length - 40)}`;
+                        let from, to;
+                        if(this.tokenType === 'erc1155'){
+                            from = `0x${log.topics[2].substring(log.topics[2].length - 40)}`;
+                            to = `0x${log.topics[3].substring(log.topics[3].length - 40)}`;
+                        } else {
+                            from = `0x${log.topics[1].substring(log.topics[1].length - 40)}`;
+                            to = `0x${log.topics[2].substring(log.topics[2].length - 40)}`;
```
[the remark this answers is not in the document]
Comment (EJSG) [reply src/components/TransferTable.vue:141]: this should do it (const name might not be perfect though):
```js
const getTopicHash = index => `0x${log.topics[index].substring(log.topics[index].length - 40)}`
```

File: src/components/TransferTable.vue [hunk 3]
```diff
@@ -190,10 +202,14 @@ export default {
         getPath(props) {
             const { page, rowsPerPage, descending } = props.pagination;
             let path = `/v2/evm/get_transactions?limit=${
-                rowsPerPage === 0 ? 500 : rowsPerPage
+                rowsPerPage === 0 ? 10 : rowsPerPage
             }`;
-
-            path += `&log_topics=${TRANSFER_EVENT_SIGNATURE},${this.address}`
+            // TODO: switch path according to ERC type (ERC20 & ERC721 have same sig)
```
Comment (EJSG) [inline src/components/TransferTable.vue:207]: TODO should reference a github issue or be resolved before PR is merged

File: src/lib/Contract.js [hunk 4]
```diff
@@ -114,7 +113,10 @@ export default class Contract {
 
     return await Promise.all(logsArray.map(async log => {
       let parsedLog = await this.parseEvent(log);
-      parsedLog = this.formatLog(log, parsedLog);
+      if(parsedLog.name && parsedLog.eventFragment && parsedLog.eventFragment.inputs){
```
Comment (EJSG) [inline src/lib/Contract.js:116]: `parsedLog.eventFragment && parsedLog.eventFragment.inputs` can be simplified to `parsedLog.eventFragment?.inputs`

File: src/lib/ContractManager.js [hunk 5]
```diff
@@ -58,15 +58,22 @@ export default class ContractManager {
...
-        token.metadata = await contract.tokenURI(tokenId);
         token.metadata = token.metadata.replace('ipfs://', 'https://cloudflare-ipfs.com/ipfs/');
 
         return token;
     }
+
+    getTokenTypeFromLog(log){
+        let sig = log.topics[0].substr(0, 10);
+        let type = (log.topics.length === 4) ? 'erc721' : 'erc20';
+        return (sig === '0xc3d58168') ? 'erc1155' : type;
```
Comment (EJSG) [inline src/lib/ContractManager.js:75]: constant for the signature would be nice for readability

**Comment by EJSG** [comment]
looks great to me! just a few comments + what don said. also do you have an example of an address with `ERC1155` xfers I can test with?

File: src/components/Transaction/InternalTxns.vue [hunk 6]
```diff
@@ -34,8 +34,8 @@ export default {
         for(let k = 0; k < this.itxs.length;k++){
             let itx = this.itxs[k];
             let contract = await this.getContract(itx.to);
-            let fnsig = itx.input.slice(0, 8);
-            let name = fnsig ? false : 'TLOS transfer';
+            let name = (itx.value)  ? 'TLOS transfer' : 'Unknown (' + '0x' + itx.input.slice(0, 8) +')';
+            name = (itx.type === 'create')  ? 'Contract deployment' : name;
```
[the remark this answers is not in the document]
Comment (tomtomcrypto) [reply src/components/Transaction/InternalTxns.vue:38]: It cannot be a const as this is the fallback, the name among other variable gets overriden later if we can parse with ABI

[the remark this answers is not in the document]
Comment (donnyquixotic) [reply src/components/Transaction/InternalTxns.vue:38]: that's fine.

File: src/components/Transaction/LogsViewer.vue [hunk 7]
```diff
@@ -76,7 +76,9 @@ export default {
             const log = this.logs[i];
             const function_signature = log.topics[0].substr(0, 10);
             if(TRANSFER_SIGNATURES.includes(function_signature)) {
-                contract = await this.getLogContract(log, (log.topics.length === 4) ? 'erc721': 'erc20');
+                let type = (log.topics.length === 4) ? 'erc721': 'erc20';
+                type = (function_signature === '0xc3d58168') ? 'erc1155' : type;
```
[the remark this answers is not in the document]
Comment (tomtomcrypto) [reply src/components/Transaction/LogsViewer.vue]: now imported from utils

**Review by EJSG — APPROVED** [review]
:fire:

**Review by Viterbo — APPROVED** [review]

**Pull request merged**